src/util/system.cpp

```diff
@@ -1011,7 +1011,18 @@ std::string ArgsManager::GetChainName() const
     if (fTestNet)
         return CBaseChainParams::TESTNET;
 
-    return GetArg("-chain", CBaseChainParams::MAIN);
+    auto chain_name = GetArg("-chain", CBaseChainParams::MAIN);
+    std::string chain_name_post_alias;
+    if (chain_name == CBaseChainParams::MAINNET_ALIAS) {
```

Comment:
Wouldn't it be better using a switch statement here?

Reply:
Old:

    if (chain_name == CBaseChainParams::MAINNET_ALIAS) {
        chain_name_post_alias = CBaseChainParams::MAIN;
    } else if (chain_name == CBaseChainParams::TESTNET_ALIAS) {
        chain_name_post_alias = CBaseChainParams::TESTNET;
    } else if (chain_name == CBaseChainParams::TESTNET3_ALIAS) {
        chain_name_post_alias = CBaseChainParams::TESTNET;
    } else {
        chain_name_post_alias = chain_name;
    }
    return chain_name_post_alias;

I would do this. What about this?
I have had second thoughts about accepting `testnet3` here.

    switch(chain_name) {
    case CBaseChainParams::MAINNET_ALIAS:
        return CBaseChainParams::MAIN;
    case CBaseChainParams::TESTNET_ALIAS:
        return CBaseChainParams::TESTNET;
    }
    return chain_name;

Reply:
I never really upgraded to C++11.

It'll take me a while to figure this out.